src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -124,16 +110,42 @@ public AggregationManager SetCollectionPeriod(TimeSpan collectionPeriod)
             Debug.Assert(collectionPeriod.TotalSeconds >= MinCollectionTimeSecs);
             lock (this)
             {
-                _collectionPeriod = collectionPeriod;
+                CollectionPeriod = collectionPeriod;
             }
             return this;
         }
 
+        private void CompletedMeasurements(Instrument instrument, object? cookie)
+        {
+            _endInstrumentMeasurements(instrument);
+            RemoveInstrumentState(instrument);
+        }
+
+        private void PublishedInstrument(Instrument instrument, MeterListener _)
+        {
+            _instrumentPublished(instrument);
+            InstrumentState? state = GetInstrumentState(instrument);
+            if (state != null)
+            {
+                _beginInstrumentMeasurements(instrument);
+
+                if (!_instrumentList.Contains(instrument))
```

Comment:
Is this list unbounded? It does not appear that we ever remove instrumented from here.

Reply:
We don't currently remove instruments from that list - I wasn't anticipating that being an issue from a scale perspective, but that could be an incorrect assumption if someone has a different perspective on it.